### PR TITLE
emscripten upstream

### PR DESCRIPTION
**Pull request by thewtex**
- COMP: Remove invalid Emscripten HDF5 link flags
- COMP: Declare H5O__fsinfo_set_version
- COMP: Provide H5CX_set_apl declaration in H5Odeprec.c
- COMP: Provide H5CX_get_ohdr_flags declaration in H5Oint.c
- COMP: Declare H5CX_set_libver_bounds in H5Rint.c
- COMP: Add output redirection for H5Tinit.c, H5lib_settings.c generation
- BUG: Disable Emscripten exception for floating point exceptions
- COMP: Include stdio.h for printf

### CODE REVIEW

**Comment by thewtex** [comment]
HDF5 and MINC patches have been contributed upstream.

File: Modules/ThirdParty/MINC/src/libminc/volume_io/Volumes/input_nifti.c [hunk 1]
```diff
@@ -10,6 +10,9 @@
 
 #include "nifti1.h"
 #include "nifti1_io.h"
+#ifdef DEBUG
```
Comment (phcerdan) [inline Modules/ThirdParty/MINC/src/libminc/volume_io/Volumes/input_nifti.c:13]: `DEBUG` or `_DEBUG` like the other file?

File: Modules/ThirdParty/HDF5/src/itkhdf5/src/CMakeLists.txt [hunk 2]
```diff
@@ -992,7 +992,7 @@ if (NOT EXISTS "${HDF5_GENERATED_SOURCE_DIR}/H5Tinit.c")
         OUTPUT ${HDF5_GENERATED_SOURCE_DIR}/H5Tinit.c
                ${HDF5_GENERATED_SOURCE_DIR}/gen_SRCS.stamp1
         COMMAND ${CMAKE_CROSSCOMPILING_EMULATOR} $<TARGET_FILE:H5detect>
-        ARGS ${HDF5_GENERATED_SOURCE_DIR}/H5Tinit.c
+        ARGS > ${HDF5_GENERATED_SOURCE_DIR}/H5Tinit.c
```
Comment (phcerdan) [inline Modules/ThirdParty/HDF5/src/itkhdf5/src/CMakeLists.txt:995]: Cool

**Review by phcerdan — APPROVED** [review]

File: Modules/ThirdParty/MINC/src/libminc/volume_io/Volumes/input_mgh.c [hunk 3]
```diff
@@ -16,6 +16,9 @@
 #endif
 #include "znzlib.h"
 #include "errno.h"
+#ifdef DEBUG
```
Comment (dzenanz) [inline Modules/ThirdParty/MINC/src/libminc/volume_io/Volumes/input_mgh.c:19]: `_DEBUG`?

File: Modules/Core/Common/src/itkFloatingPointExceptions_unix.cxx [hunk 4]
```diff
@@ -136,7 +136,7 @@ void
 FloatingPointExceptions::Enable()
 {
   itkInitGlobalsMacro(PimplGlobals);
-#if defined(ITK_HAS_FPE_CAPABILITY) && !defined(__EMSCRIPTEN__)
+#if defined(ITK_HAS_FPE_CAPABILITY)
   itk_feenableexcept(FE_DIVBYZERO);
```
Comment (dzenanz) [inline Modules/Core/Common/src/itkFloatingPointExceptions_unix.cxx:140]: What needs to be a no-op? `itk_feenableexcept` or something else?

File: Modules/ThirdParty/MINC/src/libminc/volume_io/Volumes/input_nifti.c (hunk 1 above)
Comment (thewtex) [reply Modules/ThirdParty/MINC/src/libminc/volume_io/Volumes/input_nifti.c:13]: There was some differences between the files, but I kept the pattern in upstream.

File: Modules/ThirdParty/MINC/src/libminc/volume_io/Volumes/input_mgh.c (hunk 3 above)
Comment (thewtex) [reply Modules/ThirdParty/MINC/src/libminc/volume_io/Volumes/input_mgh.c:19]: There was some differences between the files, but I kept the pattern in upstream.

File: Modules/Core/Common/src/itkFloatingPointExceptions_unix.cxx (hunk 4 above)
Comment (thewtex) [reply Modules/Core/Common/src/itkFloatingPointExceptions_unix.cxx:140]: `itk_feenableexcept` is a noop, but we needed it to entered this code path.

**Pull request merged**